src/AnalyzerVisitors/IrradiationPower.cpp

```diff
@@ -1 +1,69 @@
 #include "AnalyzerVisitors/IrradiationPower.h"
+
+void IrradiationPowerVisitor::preVisit() {
+  irradiatedPowerConsumptionSummaries.clear();   
+}
+
+void IrradiationPowerVisitor::visit(SimParms& sp) {
+  timeIntegratedLumi = sp.timeIntegratedLumi();
+  referenceTemp    = sp.referenceTemp() + insur::celcius_to_kelvin;   
+  alphaParam       = sp.alphaParam();
+  irradiationMap_  = &sp.irradiationMapsManager();
+}
+
+void IrradiationPowerVisitor::visit(Barrel& b) {
+  irradiatedPowerConsumptionSummaries[b.myid()].setHeader("Layer", "Ring");
+  irradiatedPowerConsumptionSummaries[b.myid()].setPrecision(3);        
+}
+
+void IrradiationPowerVisitor::visit(Endcap& e) {
+  irradiatedPowerConsumptionSummaries[e.myid()].setHeader("Disk", "Ring");
+  irradiatedPowerConsumptionSummaries[e.myid()].setPrecision(3);        
+}
+
+void IrradiationPowerVisitor::visit(DetectorModule& m) {
+  operatingTemp    = m.operatingTemp() + insur::celcius_to_kelvin;
+  chargeDepletionVoltage = m.chargeDepletionVoltage();
+  double volume = 0.;
+  std::vector<double> irradiationValues;
+  
+  for (const auto& s : m.sensors()) {
+    // Calculate total volume occupied by sensors
+    volume += s.sensorThickness() * m.area() / 1000.0; // volume in cm^3
```

Comment:
Please use units here: instead of /1000.0 it should be (probably) (/Unit::mm*Unit::cm)^3